pkg/workerpool/observability.go

```diff
@@ -102,6 +106,15 @@ var poolMetrics = &workPoolMetrics{
 		},
 		[]string{poolLabel},
 	),
+	workLatency: metrics.NewContextualHistogramVec(
+		prometheus.HistogramOpts{
+			Subsystem: subsystem,
+			Name:      "work_latency_seconds",
+			Help:      "Histogram of message processing latency (seconds)",
+			Buckets:   []float64{0.2, 0.4, 0.6, 0.8, 1.0, 2.0},
```

Comment:
Are these buckets appropriate? Don't we care about < 200ms?

Reply:
I'm trying to balance cardinality and usefulness. How about the following ?

`[]float64{0.05, 0.1, 0.2, 0.4, 0.6, 0.8, 1.0, 1.5, 2.0, 2.5, 3.0, 3.5, 4.0, 4.5, 5.0}`